packages/backup-restore-operator/charts/templates/resourceset.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```

Comment:
I think CRD should defines in `crds` folder?

Reply:
@StrongMonkey I think there was something wrong with adding them in crds folder, where uninstall/upgrade didn't work, and it was decided to add them in templates folder?

Reply:
@StrongMonkey I think adding crds in `crds` folder didn't take care of uninstall/upgrade, hence it was decided to put them in templates folder as per the `Helm Chart Requirements` doc:
```
Put the CRD’s in the main template folder, that way they are cleaned up on delete.
```

Reply:
yeah, but arvind did came up with a plan so that crd can be made into a separate chart.  @aiyengar2 Not sure if you were going to use this.

Reply:
I thought that was more for a chart that needed to create a CR for one of the CRDs it installs? And since backup-restore-operator doesn't need to create a CR during installation I thought it'd be fine? if not I'll change it

Reply:
My understanding is that how @mrajashree approached this implementation is correct. If you aren't creating a CR from a custom CRD, you move the CRDs to the templates folder so that can be cleaned up. 

Has that changed @aiyengar2 @cbron?

Reply:
> I thought that was more for a chart that needed to create a CR for one of the CRDs it installs? And since backup-restore-operator doesn't need to create a CR during installation I thought it'd be fine? if not I'll change it

Yes this is correct. 

@StrongMonkey the separate CRD chart is only required if you need to both install the CRDs and deploy resources of that kind into a cluster in the same chart since Helm will not be able to identify a resource of the CRD's kind before the CRD is installed to the cluster. However, if you aren't trying to deploy resources of that kind into a cluster, just keeping the CRDs in the `templates/` folder of the main chart should be sufficient.

Reply:
Question to @cbron and @deniseschannon though; since Rajashree is not using a separate CRD chart, does she need to add the annotation for `catalog.cattle.io/provides-gvr`?

Technically, this chart is providing CRDs that other charts can rely on, but has the unintended consequences of also installing a whole host of things other than CRDs. I'm assuming the answer here is no, but just wanted to call it out to confirm what the expectation should be for similar charts (I think Longhorn might fall under the same use case in https://github.com/rancher/charts/pull/526)

Reply:
Templates folder and no provides annotation is fine

Reply:
> > I thought that was more for a chart that needed to create a CR for one of the CRDs it installs? And since backup-restore-operator doesn't need to create a CR during installation I thought it'd be fine? if not I'll change it
> 
> Yes this is correct.
> 
> @StrongMonkey the separate CRD chart is only required if you need to both install the CRDs and deploy resources of that kind into a cluster in the same chart since Helm will not be able to identify a resource of the CRD's kind before the CRD is installed to the cluster. However, if you aren't trying to deploy resources of that kind into a cluster, just keeping the CRDs in the `templates/` folder of the main chart should be sufficient.
> 
> Related comment / discussion: [#526 (comment)](https://github.com/rancher/charts/pull/526#issuecomment-673114537)

This explanation has been added to the `README.md` in https://github.com/rancher/charts/pull/595 to clarify this requirement for any future contributors.